crates/host/src/wasmbus/mod.rs

```diff
@@ -1218,7 +1218,16 @@ impl Host {
                                                 let permit = permits.acquire_owned().await;
                                                 tasks.spawn(async move {
                                                     let _permit = permit;
-                                                    fut.await
+                                                    match timeout(max_execution_time, fut).await {
+                                                        Ok(result) => {
+                                                            if let Err(err) = result {
+                                                                error!(?err, "task failed");
+                                                            }
+                                                        },
+                                                        Err(err) => {
+                                                            error!(?err, "Component Handler timed out");
+                                                        },
+                                                    }
```

Comment:
```suggestion
                                                    match timeout(max_execution_time, fut).await {
                                                        Ok(Err(err)) => {
                                                            error!(?err, "component handler error");
                                                        },
                                                        Err(err) => {
                                                            error!(?err, "component handler timed out");
                                                        },
                                                        Ok(_) => {},
                                                    }
```

Here's one way you can combine the blocks together into a less-nested match!

Reply:
Bad indentation here, but here's the less-indented block:
```rust
match timeout(max_execution_time, fut).await {
    Ok(Err(err)) => {
        error!(?err, "component handler error");
    },
    Err(err) => {
        error!(?err, "component handler timed out");
    },
    Ok(_) => {},
}
```